.github/workflows/clang_tidy.yml

```diff
@@ -0,0 +1,146 @@
+name: Clang-tidy static analysis (Ubuntu 24.04, Python 3.12, Clang-18, Clang-tidy-18)
```

Comment:
Should this workflow be required for a Pull Request to merge? If yes, consider adding a `gha_overall_status` job as the last one, check other workflows for examples. If no, leave as is.

Reply:
@akashchi by the way, I have reverted the removal of the check for merge requirements in linux CC workflow (please see second commit). That change was done specifically for clang-tidy.

Reply:
If this workflow should be required for a PR to be merged, consider adding something like this as the last job:
https://github.com/openvinotoolkit/openvino/blob/1cc068392eaec8ddbf43535626e19d157e805334/.github/workflows/debian_10_arm.yml#L141

And modify the `Docker` job with: https://github.com/openvinotoolkit/openvino/blob/1cc068392eaec8ddbf43535626e19d157e805334/.github/workflows/debian_10_arm.yml#L65
to avoid incorrectly reported statuses for document-related Pull Requests.

Reply:
Done